rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -581,17 +585,19 @@ bool PlayerImpl::is_storage_completely_loaded() const
 
 bool PlayerImpl::play()
 {
-  {
-    rcpputils::unique_lock<std::mutex> is_in_playback_lk(is_in_playback_mutex_);
-    if (is_in_playback_.exchange(true)) {
-      RCLCPP_WARN_STREAM(
-        owner_->get_logger(),
-        "Trying to play() while in playback, dismissing request.");
-      progress_bar_->update(clock_->is_paused() ? PlayerStatus::PAUSED : PlayerStatus::RUNNING);
-      return false;
-    }
+  rcpputils::unique_lock<std::mutex> is_in_playback_lk(is_in_playback_mutex_);
+  if (is_in_playback_.exchange(true)) {
+    RCLCPP_WARN_STREAM(
+      owner_->get_logger(),
+      "Trying to play() while in playback, dismissing request.");
+    progress_bar_->update(clock_->is_paused() ? PlayerStatus::PAUSED : PlayerStatus::RUNNING);
+    return false;
   }
```

Comment:
More of a question: The other changes make sense, but I don't think this increase in the lock scope here (from just this scope to the whole method) is required, right? However, this method itself isn't really doing much, so I think it's fine